## Final Project Submission

Please fill out:
* Student name: 
* Student pace: self paced / part time / full time
* Scheduled project review date/time: 
* Instructor name: 
* Blog post URL:


In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib notebook
import seaborn as sns
import locale
from locale import atof

An attempt to write a for loop to process files given a list|Dict of names

In [2]:
# Your code here - remember to use markdown cells for comments as well!

# datasets_dict = {'bom.movie_gross':'bom_movie_gross','imdb.name.basics':'imdb_name_basics',
#             'imdb.title.akas':'imdb_title_akas','imdb.title.basics':'imdb_title_basics',
                
#            'imdb.title.crew':'imdb_title_crew','imdb.title.principals':'imdb_title_principals',
#             'imdb.title.ratings':'imdb_title_ratings','tmdb.movies':'tmdb_movies',
#            'tn.movie_budgets':'tn_movie_budgets'}

# datasets_csv = ['bom.movie_gross.csv.gz', 'imdb.name.basics.csv.gz', 'imdb.title.akas.csv.gz', 'imdb.title.basics.csv.gz',
#                'imdb.title.crew.csv.gz', 'imdb.title.principals.csv.gz', 'imdb.title.ratings.csv.gz', 'tmdb.movies.csv.gz',
#                'tn.movie_budgets.csv.gz']

# for filename in datasets_csv:
#     data = pd.read_csv(f'zippedData/{filename}', compression='gzip', header=0, sep=',', quotechar='"')
    

# def import_datasets(datasets):
#     for key, value in datasets.items():
#         value = pd.read_csv(f'zippedData/{key}.csv.gz', compression='gzip', 
#                                  header=0, sep=',', quotechar='"')
#         return value

### Import the datasets

These are the _9_ `CSV` datasets

In [47]:
bom_movie_gross = pd.read_csv('zippedData/bom.movie_gross.csv.gz', compression='gzip', 
                                 header=0, sep=',', quotechar='"')
imdb_name_basics = pd.read_csv('zippedData/imdb.name.basics.csv.gz', compression='gzip', 
                                 header=0, sep=',', quotechar='"')
imdb_title_akas = pd.read_csv('zippedData/imdb.title.akas.csv.gz', compression='gzip', 
                                 header=0, sep=',', quotechar='"')
imdb_title_basics = pd.read_csv('zippedData/imdb.title.basics.csv.gz', compression='gzip', 
                                 header=0, sep=',', quotechar='"')
imdb_title_crew = pd.read_csv('zippedData/imdb.title.crew.csv.gz', compression='gzip', 
                                 header=0, sep=',', quotechar='"')
imdb_title_principals = pd.read_csv('zippedData/imdb.title.principals.csv.gz', compression='gzip', 
                                 header=0, sep=',', quotechar='"')
imdb_title_ratings = pd.read_csv('zippedData/imdb.title.ratings.csv.gz', compression='gzip', 
                                 header=0, sep=',', quotechar='"')
tmdb_movies = pd.read_csv('zippedData/tmdb.movies.csv.gz', compression='gzip', 
                                 header=0, sep=',', quotechar='"')
tn_movie_budgets = pd.read_csv('zippedData/tn.movie_budgets.csv.gz', compression='gzip', 
                                 header=0, sep=',', quotechar='"')

These are the _2_ `TSV` datasets

In [48]:
rt_movie_info = pd.read_csv('zippedData/rt.movie_info.tsv.gz', compression='gzip', 
                                 header=0, sep='\t', encoding='ISO-8859-1', quotechar='"')
rt_reviews = pd.read_csv('zippedData/rt.reviews.tsv.gz', compression='gzip', 
                                 header=0, sep='\t', encoding='ISO-8859-1', quotechar='"')

In [49]:
bom_movie_gross.head()


title studio  domestic_gross  \
0                                  Toy Story 3     BV     415000000.0   
1                   Alice in Wonderland (2010)     BV     334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                    Inception     WB     292600000.0   
4                          Shrek Forever After   P/DW     238700000.0   

  foreign_gross  year  
0     652000000  2010  
1     691300000  2010  
2     664300000  2010  
3     535700000  2010  
4     513900000  2010

### Info 
#### Missing Values
* ~ 1300 missing in Foreign Gross and convert to float
* ~ 30 in domestic gross (already float)
* Studio is missing 5 values, may not be important

In [50]:
bom_movie_gross.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3387 entries, 0 to 3386
Data columns (total 5 columns):
title             3387 non-null object
studio            3382 non-null object
domestic_gross    3359 non-null float64
foreign_gross     2037 non-null object
year              3387 non-null int64
dtypes: float64(1), int64(1), object(3)
memory usage: 132.4+ KB


Set index to _movie title_

In [51]:
bom_movie_gross.set_index(['title'], inplace=True)


__Domestic Gross__ 
_Top 5_

Divide column by 1 million to makes totals easier to evaluate

In [52]:
bom_movie_gross['domestic_gross_mm'] = bom_movie_gross['domestic_gross'] / 1000000


In [58]:
bom_movie_gross.sort_values(by='domestic_gross_mm', ascending=False)[:5]


studio  domestic_gross foreign_gross  year  \
title                                                                     
Star Wars: The Force Awakens     BV     936700000.0       1,131.6  2015   
Black Panther                    BV     700100000.0     646900000  2018   
Avengers: Infinity War           BV     678800000.0       1,369.5  2018   
Jurassic World                 Uni.     652300000.0       1,019.4  2015   
Marvel's The Avengers            BV     623400000.0     895500000  2012   

                              domestic_gross_mm  
title                                            
Star Wars: The Force Awakens              936.7  
Black Panther                             700.1  
Avengers: Infinity War                    678.8  
Jurassic World                            652.3  
Marvel's The Avengers                     623.4

If over 1 billion, total is annotated as such `"1,131.6"` see foreign gross for Star Wars from [Box Office Mojo](https://www.boxofficemojo.com/title/tt2488496/?ref_=bo_se_r_8).

__Foreign Gross__

_Cannot sort because dtype is object_

Additionally, scale needs to be adjusted, to the same way `domestic_gross` is annotated, so that data can be analyzed on the same scale. 

In [54]:
bom_movie_gross.sort_values(by='foreign_gross', ascending=False)[:5]


studio  domestic_gross foreign_gross  year  domestic_gross_mm
title                                                                         
Life's a Breeze   Magn.          3500.0         99700  2014             0.0035
The East           FoxS       2300000.0         99700  2013             2.3000
White Lion        Scre.             NaN         99600  2010                NaN
A United Kingdom   FoxS       3900000.0       9900000  2017             3.9000
Le Havre           Jan.        612000.0       9900000  2011             0.6120

_These are the 5 films that had foreign gross > 1 billion_

In [55]:
# filter foreign gross over 1 billion
bill_filt = bom_movie_gross['foreign_gross'].str.contains(',', na=False)
bom_movie_gross[bill_filt]


studio  domestic_gross foreign_gross  year  \
title                                                                     
Star Wars: The Force Awakens     BV     936700000.0       1,131.6  2015   
Jurassic World                 Uni.     652300000.0       1,019.4  2015   
Furious 7                      Uni.     353000000.0       1,163.0  2015   
The Fate of the Furious        Uni.     226000000.0       1,010.0  2017   
Avengers: Infinity War           BV     678800000.0       1,369.5  2018   

                              domestic_gross_mm  
title                                            
Star Wars: The Force Awakens              936.7  
Jurassic World                            652.3  
Furious 7                                 353.0  
The Fate of the Furious                   226.0  
Avengers: Infinity War                    678.8

Perhaps create .py function that takes in a string (object) if it has commas, removes them, if it has decimals convert it to float and multipy by 10000000. If it has no decimals just convert to float

Once column is converted to floats all in the same decimal level, then they can be divided by 1 million to get correct scalling of foreign_gross. 

This function could be called convert_obj_float.

In [59]:
bom_movie_gross[bom_movie_gross['foreign_gross'].replace(',','', regex=True).astype(float)

TypeError: astype_nansafe() got an unexpected keyword argument 'inplace'

In [31]:
bom_movie_gross['foreign_gross'].apply(lambda num: int("".join(x for x in num if x.isnumeric())))

TypeError: 'float' object is not iterable

In [22]:
imdb_name_basics.head()


nconst       primary_name  birth_year  death_year  \
0  nm0061671  Mary Ellen Bauder         NaN         NaN   
1  nm0061865       Joseph Bauer         NaN         NaN   
2  nm0062070         Bruce Baum         NaN         NaN   
3  nm0062195       Axel Baumann         NaN         NaN   
4  nm0062798        Pete Baxter         NaN         NaN   

                                 primary_profession  \
0         miscellaneous,production_manager,producer   
1        composer,music_department,sound_department   
2                        miscellaneous,actor,writer   
3  camera_department,cinematographer,art_department   
4  production_designer,art_department,set_decorator   

                          known_for_titles  
0  tt0837562,tt2398241,tt0844471,tt0118553  
1  tt0896534,tt6791238,tt0287072,tt1682940  
2  tt1470654,tt0363631,tt0104030,tt0102898  
3  tt0114371,tt2004304,tt1618448,tt1224387  
4  tt0452644,tt0452692,tt3458030,tt2178256

In [23]:
imdb_title_akas.head()

title_id  ordering                                    title region  \
0  tt0369610        10                            Джурасик свят     BG   
1  tt0369610        11                        Jurashikku warudo     JP   
2  tt0369610        12  Jurassic World: O Mundo dos Dinossauros     BR   
3  tt0369610        13                  O Mundo dos Dinossauros     BR   
4  tt0369610        14                           Jurassic World     FR   

  language        types   attributes  is_original_title  
0       bg          NaN          NaN                0.0  
1      NaN  imdbDisplay          NaN                0.0  
2      NaN  imdbDisplay          NaN                0.0  
3      NaN          NaN  short title                0.0  
4      NaN  imdbDisplay          NaN                0.0

In [24]:
imdb_title_basics.head()

tconst                    primary_title              original_title  \
0  tt0063540                        Sunghursh                   Sunghursh   
1  tt0066787  One Day Before the Rainy Season             Ashad Ka Ek Din   
2  tt0069049       The Other Side of the Wind  The Other Side of the Wind   
3  tt0069204                  Sabse Bada Sukh             Sabse Bada Sukh   
4  tt0100275         The Wandering Soap Opera       La Telenovela Errante   

   start_year  runtime_minutes                genres  
0        2013            175.0    Action,Crime,Drama  
1        2019            114.0       Biography,Drama  
2        2018            122.0                 Drama  
3        2018              NaN          Comedy,Drama  
4        2017             80.0  Comedy,Drama,Fantasy

In [25]:
imdb_title_crew.head()

tconst                      directors              writers
0  tt0285252                      nm0899854            nm0899854
1  tt0438973                            NaN  nm0175726,nm1802864
2  tt0462036                      nm1940585            nm1940585
3  tt0835418                      nm0151540  nm0310087,nm0841532
4  tt0878654  nm0089502,nm2291498,nm2292011            nm0284943

In [26]:
imdb_title_principals.head()

tconst  ordering     nconst  category       job        characters
0  tt0111414         1  nm0246005     actor       NaN       ["The Man"]
1  tt0111414         2  nm0398271  director       NaN               NaN
2  tt0111414         3  nm3739909  producer  producer               NaN
3  tt0323808        10  nm0059247    editor       NaN               NaN
4  tt0323808         1  nm3579312   actress       NaN  ["Beth Boothby"]

In [27]:
imdb_title_ratings.head()

tconst  averagerating  numvotes
0  tt10356526            8.3        31
1  tt10384606            8.9       559
2   tt1042974            6.4        20
3   tt1043726            4.2     50352
4   tt1060240            6.5        21

In [28]:
tmdb_movies.head()

Unnamed: 0            genre_ids     id original_language  \
0           0      [12, 14, 10751]  12444                en   
1           1  [14, 12, 16, 10751]  10191                en   
2           2        [12, 28, 878]  10138                en   
3           3      [16, 35, 10751]    862                en   
4           4        [28, 878, 12]  27205                en   

                                 original_title  popularity release_date  \
0  Harry Potter and the Deathly Hallows: Part 1      33.533   2010-11-19   
1                      How to Train Your Dragon      28.734   2010-03-26   
2                                    Iron Man 2      28.515   2010-05-07   
3                                     Toy Story      28.005   1995-11-22   
4                                     Inception      27.920   2010-07-16   

                                          title  vote_average  vote_count  
0  Harry Potter and the Deathly Hallows: Part 1           7.7       10788  
1                      How to Train Your Dragon           7.7        7610  
2                                    Iron Man 2           6.8       12368  
3                                     Toy Story           7.9       10174  
4                                     Inception           8.3       22186

In [ ]:
tn_movie_budgets.head()

In [ ]:
tn_movie_budgets.info()

In [ ]:
rt_movie_info.head()

In [ ]:
rt_movie_info.info()

In [ ]:
rt_reviews.head()